### PR TITLE
rockchip: add FriendlyElec NanoPi R6C and fixes for the NanoPi R6S

### PR DESCRIPTION
**Pull request by antnyfls**
1- This add support for the FriendlyElec NanoPi R6C 
2 Fixes for the NanoPi R6S pending on the kernel mailing list'
https://lore.kernel.org/all/20240612205056.397204-1-seb-dev@mail.de

### CODE REVIEW

**Comment by odkrys** [comment]
I created an r6s image with this PR and tested it. I don't know exactly why, but it seems to have fixed the port swap issue. It seems like the pcie ethernets boot in order.

https://github.com/openwrt/openwrt/issues/16271

And the usb3.0 port is also working.

**Comment by potus002** [comment]
keep up the great work

**Comment by blocktrron** [comment]
Looks good. For the U-Boot patches i assume they have been applied upstream? If so, can you add the version it is / will first be included in? You already did this for the target patch which is not pending.

This makes it easier to identify which patches to drop when updating the base version.

**Comment by antnyfls** [comment]
> Looks good. For the U-Boot patches i assume they have been applied upstream? If so, can you add the version it is / will first be included in? You already did this for the target patch which is not pending.
> 
> This makes it easier to identify which patches to drop when updating the base version.

I did include the version on the new patch
`107-v2024.10-rc1-board-rockchip-Add-FriendlyElec-NanoPi-R6C.patch`
patch `001-backport-upstream-dts-sync.patch` is already applied, I only added the r6c DTS .

**Comment by antnyfls** [comment]
I added the version in all u boot-rockchip patches, non of them have it right now, in a new patch.

**Comment by dick902** [comment]
Tested with Nano-Pi R6C, everything ok for me.

**Review by dick902 — APPROVED** [review]

**Comment by gSpotx2f** [comment]
I tested with NanoPC R6S. It looks like the 2.5G ports issue (non-deterministic interface naming) is fixed. USB3 port works fine.
But there is a small issue with the [new LED names](https://lore.kernel.org/all/20240612205056.397204-4-seb-dev@mail.de/) in `/etc/board.d/01_leds`:

    --- a/target/linux/rockchip/armv8/base-files/etc/board.d/01_leds
    +++ b/target/linux/rockchip/armv8/base-files/etc/board.d/01_leds
    @@ -29,13 +29,13 @@ friendlyarm,nanopi-r5s)
            ucidef_set_led_netdev "lan2" "LAN2" "green:lan-2" "eth2"
            ;;
     friendlyarm,nanopi-r6c)
    -       ucidef_set_led_netdev "wan" "WAN" "wan_led" "eth0"
    -       ucidef_set_led_netdev "lan1" "LAN1" "lan1_led" "eth1"
    +       ucidef_set_led_netdev "wan" "WAN" "green:wan" "eth0"
    +       ucidef_set_led_netdev "lan1" "LAN1" "green:lan-1" "eth1"
            ;;
     friendlyarm,nanopi-r6s)
    -       ucidef_set_led_netdev "wan" "WAN" "wan_led" "eth1"
    -       ucidef_set_led_netdev "lan1" "LAN1" "lan1_led" "eth2"
    -       ucidef_set_led_netdev "lan2" "LAN2" "lan2_led" "eth0"
    +       ucidef_set_led_netdev "wan" "WAN" "green:wan" "eth1"
    +       ucidef_set_led_netdev "lan1" "LAN1" "green:lan-1" "eth2"
    +       ucidef_set_led_netdev "lan2" "LAN2" "green:lan-2" "eth0"
            ;;
     esac

**Comment by antnyfls** [comment]
> But there is a small issue with the [new LED names](https://lore.kernel.org/all/20240612205056.397204-4-seb-dev@mail.de/) in `/etc/board.d/01_leds`:

@gSpotx2f thank you for reporting, after reset my device I noticed the LED problem, Tested only the R6C.

**Review by bigsmile74 — APPROVED** [review]
Build the image today for the R6C using this PR. Wrote it to sdcard and it boots. I browsed through it a bit and it seems to work fine. Installed luci and that seems to work fine too. NICs seem in order. Bootlog was clean afaict.

I like it :smile:

File: target/linux/rockchip/patches-6.6/400-1-arm64-dts-rockchip-Add-common-definitions-for-NanoPi.patch [hunk 1]
```diff
@@ -0,0 +1,1573 @@
+From 93be10e4e180eeac9b6a57a4363e08c35bee3af9 Mon Sep 17 00:00:00 2001
+From: Sebastian Kropatsch <seb-dev@mail.de>
```
Comment (hauke) [inline target/linux/rockchip/patches-6.6/400-1-arm64-dts-rockchip-Add-common-definitions-for-NanoPi.patch:2]: Is someone working on getting these patches accepted upstream?
I can not find them in linux-next.

Comment (antnyfls) [reply target/linux/rockchip/patches-6.6/400-1-arm64-dts-rockchip-Add-common-definitions-for-NanoPi.patch:2]: > Is someone working on getting these patches accepted upstream?
> I can not find them in linux-next.

have not been merged yet, they are being reviewed https://lore.kernel.org/all/?q=nanopi-r6

Comment (hauke) [reply target/linux/rockchip/patches-6.6/400-1-arm64-dts-rockchip-Add-common-definitions-for-NanoPi.patch:2]: The patches were send about 2.5 months ago. Do you know if a updated version will be send soon?

Comment (antnyfls) [reply target/linux/rockchip/patches-6.6/400-1-arm64-dts-rockchip-Add-common-definitions-for-NanoPi.patch:2]: I don't have that info, if is strictly a  requirement that the patches be in the kernel tree I can drop `rockchip: fixes and improve for NanoPi R6 series` and `rockchip: rework LED configurations for the NanoPi R6C/R6S`  for later if they get accepted. is OK that?

Comment (hauke) [reply target/linux/rockchip/patches-6.6/400-1-arm64-dts-rockchip-Add-common-definitions-for-NanoPi.patch:2]: The patches are not looking so bad, I applied your PR.

Please send a new PR when these patches are getting merged upstream.

**Comment by hariskar** [comment]
I just got my nanopi R6S, installed OpenWrt 24.10.0 r28427-6df0e3d02a / LuCI openwrt-24.10 branch 25.014.55016~7046a1c but I have problem with the ports: If I assign eth2 to WAN interface (eth2 is preconfigured in WAN) it correspondents to the 2nd physical port whuch is LAN1. So I have to put the WAN ethernet cable to LAN1 port instead of the WAN port. Any solution?

**Comment by aenertia** [comment]
Random port swaps are happening still for me with nanopc-t6 with 24.10.2 still. I have burned mac address's permanently with the ratool ( https://github.com/redchenjs/rtnicpg ) thinking it would solve the issue permanently - but obviously the dtb enumeration is still being referenced rather than MAC somewhere during init.

**Pull request merged**